src/lib/error/api-error.ts

```diff
@@ -0,0 +1,270 @@
+import { v4 as uuidV4 } from 'uuid';
+import { FromSchema } from 'json-schema-to-ts';
+
+const UnleashApiErrorTypes = [
+    'OwaspValidationError',
+    'PasswordUndefinedError',
+    'InvalidTokenError',
+    'NoAccessError',
+    'UsedTokenError',
+    'InvalidOperationError',
+    'IncompatibleProjectError',
+    'OperationDeniedError',
+    'NotFoundError',
+    'NameExistsError',
+    'FeatureHasTagError',
+    'RoleInUseError',
+    'ProjectWithoutOwnerError',
+    'UnknownError',
+    'PasswordMismatch',
+    'PasswordMismatchError',
+    'DisabledError',
+    'ContentTypeError',
+    'NotImplementedError',
+
+    // server errors; not the end user's fault
+    'InternalError',
+] as const;
+
+// types that have extra data associated with them
+const UnleashApiErrorTypes2 = [
+    'MinimumOneEnvironmentError',
+    'BadDataError',
+    'BadRequestError',
+    'ValidationError',
+    'AuthenticationRequired',
+    'NoAccessError',
+] as const;
+
+const AllUnleashApiErrorTypes = [
+    ...UnleashApiErrorTypes,
+    ...UnleashApiErrorTypes2,
+] as const;
+
+type UnleashApiErrorKind =
```

Comment:
can it be `typeof AllUnleashApiErrorTypes[number]` since you already did the merging in lines 39-42?

Reply:
Yeah, probably! Good shout.

Reply:
Addressed in 9ae66c505874bbd38d5116b755e88920fbf2341b